src/csharp/Intel/Iced/Intel/Assembler/AssemblerMemoryOperandFactory.cs

```diff
@@ -52,7 +55,7 @@ namespace Iced.Intel {
 		/// <summary>
 		/// Size of this memory operand.
 		/// </summary>
-		public readonly MemoryOperandSize Size;
+		internal readonly MemoryOperandSize Size;
```

Comment:
`Prefix` field below and any other public fields should also be internal, or is there a good reason to keep public?

Reply:
I can make them internal.

Reply:
Hm actually, I'm a bit annoyed to make them internal. If you have an assembler memory operand on the side and you want to automate change to it (modify displacement based on it, or query for the register...), it would no longer be possible.

Reply:
But they're readonly so you can't change them?

Reply:
yeah, well, now that MemoryOperandSize and AssemblerOperandFlags re internal, and there is no public constructors... it's difficult... I can make them internal then and we will see if some folks are annoyed by having them internal.

Reply:
You can create them using the normal __[] syntax, that should be enough right?

Reply:
Yes it should be ok